properties/base.py

```diff
@@ -182,14 +182,18 @@ def _notify(self, change):
             listener.func(self, change)
 
     def _set(self, name, value):
-        out = self._notify(dict(name=name, value=value, mode='validate'))
-        if out is not None:
-            value = out
-        if value is basic.undefined and name in self._backend:
+        change = dict(name=name, value=value, mode='validate')
+        self._notify(change)
+        if change['name'] != name:
+            warn('Specified Property for assignment changed during '
```

Comment:
Ah, I read that as `change['value']`, I think an exception would be better here?